### PR TITLE
Add support for NVIDIA HPC SDK Compilers on ARM

### PR DESCRIPTION
**Pull request by JTOFH**
Add option to compile WRF on ARM architecture (aarch64) systems with the NVIDIA HPC SDK compilers

TYPE: enhancement

KEYWORDS: NVHPC, nvfortran, nvc, compile, aarch64

SOURCE: Jeff Adie (NVIDIA Corporation)

DESCRIPTION OF CHANGES:
Problem:
No support for NVHPC compilers on ARM

Solution:
Added stanza to configure.defaults to add new compilation option

LIST OF MODIFIED FILES:
arch/configure.defaults

TESTS CONDUCTED: 
1. Tested successfully compiled for multiple versions of NVHPC (21.9, 22.1, 22.7)
2. It doesn't affect other part of the code. It passed regression tests.

RELEASE NOTE: Added support to build WRF on ARM64 (aarch64) architecture systems with the NVIDIA HPC SDK Compilers (nvfortran,nvc)

### CODE REVIEW

**Comment by dudhia** [comment]
Better to change the target to the develop branch.

**Comment by weiwangncar** [comment]
@JTOFH @dudhia I have changed the base branch. And it has no impact to our code, and the reg tests have passed.

**Comment by weiwangncar** [comment]
@JTOFH @dudhia When I changed the base branch, the new changes for 4.4.1 all showed up in the changed files. There may be a better way to do this....

**Comment by dudhia** [comment]
I see. That merge back to develop should happen soon as the release is already out.

**Comment by weiwangncar** [comment]
Jenkins tests:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           58
    Number of Simulations  : 158           156        0
    Number of Comparisons  : 95           92        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Review by dudhia — DISMISSED** [review]
for develop (V4.5)

File: arch/configure.defaults
Comment (mgduda) [inline arch/configure.defaults:2072]: It looks to me like this would for `x86_64` as well as for `aarch64`. If you agree, we could change this line to something like:
```
#ARCH    Linux aarch64 x86_64, NVHPC compiler with nvc # serial smpar dmpar dm+sm
```

File: arch/configure.defaults
Comment (mgduda) [inline arch/configure.defaults:2094]: Would all of these options be applicable to `aarch64` if a user were to uncomment them?

**Comment by weiwangncar** [comment]
@JTOFH Can you reply to the comment from @mgduda?

File: arch/configure.defaults
Comment (JTOFH) [reply arch/configure.defaults:2072]: That's correct - today! But there may be aarch64 specific flags we would like to add in the future so my preference would be to keep them separate.

File: arch/configure.defaults
Comment (JTOFH) [reply arch/configure.defaults:2094]: Mostly yes, other than fastsse which is x86 specific. Msmartalloc seems to be deprecated.  The other flags are OK,
The -fastsse flag equivalent is -fast.

File: arch/configure.defaults
Comment (mgduda) [reply arch/configure.defaults:2072]: That sounds reasonable -- thanks for clarifying.

File: arch/configure.defaults
Comment (mgduda) [reply arch/configure.defaults:2094]: In this case, I'd suggest cleaning up these flags by removing deprecated flags or flags that don't apply to aarch64. Essentially, I think everything in this new stanza should be tailored to aarch64.

**Comment by weiwangncar** [comment]
@JTOFH Can you reply to the comment from mgduda? You will also need to resolve the conflict since your post. Thanks!

File: arch/configure.defaults
Comment (JTOFH) [reply arch/configure.defaults]: I cleaned up and tested the flags on ARM. Pushed a new version. I'm keeping fprelaxed there but commented out as it does  change the output slightly (at least over a 24 hour run).

**Comment by weiwangncar** [comment]
The Jenkins test results:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by weiwangncar** [comment]
@mgduda Can you take another look at this PR? Thanks!

**Review by mgduda — APPROVED** [review]
Looks good to me!

**Comment by weiwangncar** [comment]
The Jenkins test results after resolving conflicts:

    Test Type              | Expected  | Received |  Failed
    = = = = = = = = = = = = = = = = = = = = = = = =  = = = =
    Number of Tests        : 23           24
    Number of Builds       : 60           57
    Number of Simulations  : 158           150        0
    Number of Comparisons  : 95           86        0

    Failed Simulations are: 
    None
    Which comparisons are not bit-for-bit: 
    None

**Comment by weiwangncar** [comment]
@mgduda @dudhia Can you approve this PR again? After resolving the conflicts, the approvals are lost.

File: arch/configure.defaults
Comment (mgduda) [inline arch/configure.defaults:2112]: @weiwangncar Was the introduction of this whitespace change intentional?

**Review by dudhia — APPROVED** [review]

File: arch/configure.defaults
Comment (weiwangncar) [reply arch/configure.defaults:2112]: @mgduda Definitely not intentional. It was cut and pasted. Let's see if it can be removed.

**Review by mgduda — APPROVED** [review]

File: arch/configure.defaults
Comment (mgduda) [reply arch/configure.defaults]: Looks good now. Thanks!

**Review by weiwangncar — APPROVED** [review]
Approve it for Jimy.

**Pull request merged**
